registry/remote/credentials/memory_store.go

```diff
@@ -32,6 +35,30 @@ func NewMemoryStore() Store {
 	return &memoryStore{}
 }
 
+// NewMemoryStoreFromDockerConfig creates a new in-memory credentials store from the given configuration.
+//
+// Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+func NewMemoryStoreFromDockerConfig(c []byte) (*memoryStore, error) {
```

Comment:
Public method should return public type
```suggestion
func NewMemoryStoreFromDockerConfig(c []byte) (Store, error) {
```

Reply:
Done